routing/router.go

```diff
@@ -1473,7 +1477,13 @@ func (r *ChannelRouter) assertNodeAnnFreshness(node route.Vertex,
 		return errors.Errorf("unable to query for the "+
 			"existence of node: %v", err)
 	}
-	if !exists {
+
```

Comment:
The comment above isn't fully valid anymore:
```
// If we are not already aware of this node, it means that we don't
// know about any channel using this node.
```
IIUC we check with the switch, exactly for the case where we don't know about the node but *do* know about a channel using this node.

I suggest to change the comment to better explain this.